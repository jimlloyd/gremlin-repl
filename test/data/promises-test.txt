// Simple examples of using Q promises in a REPL script.
var d = Q.defer()
d.resolve('testing 1,2,3.')
d.promise

var d1 = Q.defer()
var d2 = Q.defer()
var timers = require('timers')
var ignore1 = timers.setTimeout(function() { d1.resolve(23); }, 60)
var ignore2 = timers.setTimeout(function() { d2.resolve(42); }, 50)
Q.all([d1.promise, d2.promise])

