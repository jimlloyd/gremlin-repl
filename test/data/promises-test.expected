node > // Simple examples of using Q promises in a REPL script.
node > var d = Q.defer()
node > d.resolve('testing 1,2,3.')
node > d.promise
'testing 1,2,3.'
node > 
node > var d1 = Q.defer()
node > var d2 = Q.defer()
node > var timers = require('timers')
node > var ignore1 = timers.setTimeout(function() { d1.resolve(23); }, 60)
node > var ignore2 = timers.setTimeout(function() { d2.resolve(42); }, 50)
node > Q.all([d1.promise, d2.promise])
[ 23, 42 ]
node > 
